tests/test_scalars.py

```diff
@@ -24,17 +39,685 @@
+
+    lt_a = check(
+        assert_type(c_timedelta_index < td, np_ndarray_bool), np.ndarray, np.bool_
+    )
+    ge_a = check(
+        assert_type(c_timedelta_index >= td, np_ndarray_bool), np.ndarray, np.bool_
+    )
+    assert (lt_a != ge_a).all()
+
+    eq_s = check(
+        assert_type(c_timedelta_series < td, "pd.Series[bool]"), pd.Series, bool
+    )
+    ne_s = check(
+        assert_type(c_timedelta_series >= td, "pd.Series[bool]"), pd.Series, bool
+    )
+    assert (eq_s != ne_s).all()
+
+    eq = check(assert_type(td == td, bool), bool)
+    ne = check(assert_type(td != td, bool), bool)
+    assert eq != ne
+
+    eq = check(assert_type(td == c_dt_timedelta, bool), bool)
+    ne = check(assert_type(td != c_dt_timedelta, bool), bool)
+    assert eq != ne
+
+    eq = check(assert_type(td == c_timedelta64, bool), bool)
+    ne = check(assert_type(td != c_timedelta64, bool), bool)
+    assert eq != ne
+
+    eq_a = check(
+        assert_type(td == c_ndarray_td64, np_ndarray_bool), np.ndarray, np.bool_
+    )
+    ne_a = check(
+        assert_type(td != c_ndarray_td64, np_ndarray_bool), np.ndarray, np.bool_
+    )
+    assert (eq_a != ne_a).all()
+
+    eq_a = check(
+        assert_type(td == c_timedelta_index, np_ndarray_bool), np.ndarray, np.bool_
+    )
+    ne_a = check(
+        assert_type(td != c_timedelta_index, np_ndarray_bool), np.ndarray, np.bool_
+    )
+    assert (eq_a != ne_a).all()
+
+    eq_s = check(
+        assert_type(td == c_timedelta_series, "pd.Series[bool]"), pd.Series, bool
+    )
+    ne_s = check(
+        assert_type(td != c_timedelta_series, "pd.Series[bool]"), pd.Series, bool
+    )
+    assert (eq_s != ne_s).all()
+
+    eq = check(assert_type(td == 1, Literal[False]), bool)
+    ne = check(assert_type(td != 1, Literal[True]), bool)
+    assert eq != ne
+
+    eq = check(assert_type(td == (3 + 2j), Literal[False]), bool)
+    ne = check(assert_type(td != (3 + 2j), Literal[True]), bool)
+    assert eq != ne
+
+
+def test_timedelta_cmp_rhs() -> None:
+    # Test that check eq and ne when Timedelta is the RHS argument
+    # that use the __eq__ and __ne__ methods of the LHS
+    td = pd.Timedelta("1 day")
+    ndarray_td64: npt.NDArray[np.timedelta64] = np.array(
+        [1, 2, 3], dtype="timedelta64[D]"
+    )
+    c_dt_timedelta = dt.timedelta(days=1)
+    c_timedelta64 = np.timedelta64(1, "D")
+    c_ndarray_td64 = ndarray_td64
+    c_timedelta_index = pd.TimedeltaIndex([1, 2, 3], unit="D")
+    c_timedelta_series = pd.Series(pd.TimedeltaIndex([1, 2, 3]))
+
+    eq = check(assert_type(c_dt_timedelta == td, bool), bool)
+    ne = check(assert_type(c_dt_timedelta != td, bool), bool)
+    assert eq != ne
+
+    eq = check(assert_type(c_timedelta64 == td, Any), np.bool_)
+    ne = check(assert_type(c_timedelta64 != td, Any), np.bool_)
+    assert eq != ne
+
+    eq_a = check(assert_type(c_ndarray_td64 == td, Any), np.ndarray)
+    ne_a = check(assert_type(c_ndarray_td64 != td, Any), np.ndarray)
```

Comment:
for the above 2 tests, add `np.bool_` to the `check` arguments

Reply:
Fixed.